### DISCUSSION TITLE
when using undertow,how should i config Trace Profilling endpoint

### DISCUSSION THREAD

**Discussion by wangan0915** [apache/skywalking#10062] [Q&A] [answered]
In SpringBoot, When undertow is used to replace tomcat,Trace Profilling can not get any data if endpoint is /api/getCommonDataWithTemplateId, but when endpoint is UndertowDispatch,i can get data.
i think right endpoint is /api/getCommonDataWithTemplateId rather than UndertowDispatch when i using Trace Profilling
![f635d097bd1247b1350c0031e246ad1](https://user-images.githubusercontent.com/39958329/204767249-2ef00977-d1a1-4505-8e99-863671eae928.png)
![8fbd2a0269f8be3864b7f4a1ed5bebb](https://user-images.githubusercontent.com/39958329/204767286-8845c27e-5f94-410e-846f-e5b05210e218.png)
![1cda35ffcfe43e00e6b9dec69908069](https://user-images.githubusercontent.com/39958329/204767295-6a0e0530-0bea-4f79-8242-aa3165ba972e.png)

**Comment by wu-sheng** [comment, marked as answer]
Java agent's profiling is segment-oriented. So, what you described is expected from a design perspective. 

From what I am reading from your traces, your traffic statistics should be double-counted. One request should only have one entry span, you seem to have two, one from the undertow, and the other from SpringMVC.

Consider to disable undertow plugin and choose SpringMVC and its endpoint name
